common/djangoapps/dark_lang/tests.py

```diff
@@ -229,78 +228,120 @@ def test_partial_match_es_la(self, latin_america_code):
 
         self.assertAcceptEquals(
             'es-419;q=1.0',
-            self.process_request(accept='{};q=1.0, pt;q=0.5'.format(latin_america_code))
+            self.process_middleware_request(accept='{};q=1.0, pt;q=0.5'.format(latin_america_code))
         )
 
-    def assertSessionLangEquals(self, value, request):
+    def assert_request_session_lang_equals(self, value, request):
+        """
+        Assert that the LANGUAGE_SESSION_KEY set in request.session is equal to value
+        """
+        self.assert_session_lang_equals(
+            value,
+            request.session
+        )
+
+    def assert_session_lang_equals(self, value, session):
         """
         Assert that the LANGUAGE_SESSION_KEY set in request.session is equal to value
         """
         self.assertEquals(
             value,
-            request.session.get(LANGUAGE_SESSION_KEY, UNSET)
+            session.get(LANGUAGE_SESSION_KEY, UNSET)
         )
 
+    def post_set_preview_lang(self, preview_language):
+        return self.client.post('/update_lang/', {'preview_lang': preview_language, 'set_language': 'set_language'})
+
+    def post_clear_preview_lang(self):
+        return self.client.post('/update_lang/', {'reset': 'reset'})
+
     def test_preview_lang_with_released_language(self):
         # Preview lang should always override selection.
-        self.assertSessionLangEquals(
+        self.post_set_preview_lang('rel')
+        response = self.client.get('/home')
+        self.assert_session_lang_equals(
             'rel',
-            self.process_request(preview_lang='rel')
+            response.client.session
         )
 
-        self.assertSessionLangEquals(
+        # post_request = self.process_darklang_post_request(preview_lang='rel', language_session_key='notrel')
+        self.post_set_preview_lang('notrel')
+        self.post_set_preview_lang('rel')
+        response = self.client.get('/home')
+        self.assert_session_lang_equals(
             'rel',
-            self.process_request(preview_lang='rel', language_session_key='notrel')
+            response.client.session
         )
 
     def test_preview_lang_with_dark_language(self):
-        self.assertSessionLangEquals(
+        self.post_set_preview_lang('unrel')
+        response = self.client.get('/home')
+        self.assert_session_lang_equals(
             'unrel',
-            self.process_request(preview_lang='unrel')
+            response.client.session
         )
 
-        self.assertSessionLangEquals(
+        # Test a clear and then a set of the preview language
+        self.post_clear_preview_lang()
```

Comment:
Again, should you assert something after clearing the language? As it is, I'm not sure what is being tested because the language was unrel before the clear.